src/Go/boardAnalysis/goAI.ts

```diff
@@ -37,12 +37,11 @@ export function makeAIMove(boardState: BoardState, useOfflineCycles = true): Pro
 
   // If the AI is disabled, simply make a promise to be resolved once the player makes a move as white
   if (boardState.ai === GoOpponent.none) {
-    GoEvents.emit();
-    // Update currentTurnResolver to call Go.nextTurn's resolve function with the last played move's details
-    Go.nextTurn = new Promise((resolve) => (currentTurnResolver = () => resolve(getPreviousMoveDetails())));
+    resetAI();
   }
   // If an AI is in use, find the faction's move in response, and resolve the Go.nextTurn promise once it is found and played.
   else {
+    const currentMoveCount = Go.currentGame.previousBoards.length;
     Go.nextTurn = getMove(boardState, GoColor.white, Go.currentGame.ai, useOfflineCycles).then(
       async (play): Promise<Play> => {
         if (boardState !== Go.currentGame) return play; //Stale game
```

Comment:
You already have a check here. How does it differ from checking the current move count? Does one check cover more or less than the other?

Reply:
The boardState check is just object pointer comparison, it just looks for entirely new games. The move count is to make sure nothing has changed in the current game due to potential race conditions. I'd like to think there aren't any that could trigger this, but it doesn't hurt. I unified the checks made further down after some calls have been awaited, but the move count couldn't have changed yet in this highlighted line.